pynsot/commands/cmd_protocol_types.py

```diff
@@ -0,0 +1,340 @@
+    """
+    List existing Protocol Types for a Site.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When listing Protocol Types, all objects are displayed by default. You
+    optionally may lookup a single Protocol Types by ID using the -i/--id option.
+    The ID can either be the numeric ID of the Protocol Type.
+    """
+    data = ctx.params
+
+    # If we provide ID, show more fields
+    if id is not None or name:
+        display_fields = VERBOSE_FIELDS
+    else:
+        display_fields = DISPLAY_FIELDS
+
+    # If we aren't passing a sub-command, just call list(), otherwise let it
+    # fallback to default behavior.
+    if ctx.invoked_subcommand is None:
+        ctx.obj.list(
+            data, display_fields=display_fields,
+            verbose_fields=VERBOSE_FIELDS
+        )
+
+@list.command()
+@click.pass_context
+def protocols(ctx, *args, **kwargs):
+    """Recursively get all protocols of an Protocol Type."""
+    callbacks.list_subcommand(
+        ctx, display_fields=PROTOCOL_DISPLAY_FIELDS, my_name=ctx.info_name
+    )
+
+
+# Remove
+@cli.command()
+@click.option(
+    '-I',
+    '--id',
+    metavar='ID',
+    help='Unique ID of the Protocol Type being deleted.',
+    required=True,
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol Type is under.',
+    callback=callbacks.process_site_id,
+    required=True,
+)
+@click.pass_context
+def remove(ctx, id, site_id):
+    """
+    Remove an Protocol Type.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When removing an Protocol Type, you must provide the ID of the Protocol Type using
+    -i/--id.
+
+    You may retrieve the ID for an Protocol Type by parsing it from the list of
+    Protocol Types for a given Site:
+
+        nsot protocol_types list --site-id <site_id> | grep <protocol_type name>
+    """
+    data = ctx.params
+    ctx.obj.remove(**data)
+
+
+# Update
+@cli.command()
+@click.option(
+    '-r',
+    '--required_attributes',
+    metavar='REQUIRED ATTRIBUTES',
+    help='A key/value pair attached to this Protocol Type (format: key=value).',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-e',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='The description for this Protocol Type.',
+)
+@click.option(
+    '-I',
+    '--id',
+    metavar='ID',
+    type=types.NATURAL_KEY,
+    help='Unique ID of the Protocol Type being updated.',
+    required=True,
+)
+@click.option(
+    '-n',
+    '--name',
+    metavar='NAME',
+    type=str,
+    help='The name of the Protocol Type.',
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Protocol Type is under.',
+    callback=callbacks.process_site_id,
+    required=True,
+)
+@click.option(
```

Comment:
Lines 283-311 are not needed for this object since it can't have attributes. :) Or were you using this to handle the manipulation of the existing required attributes? If so, the help text for each flag needs to be updated to reflect that, as well as the help text for the `update` command.